src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,276 @@
+            }
+            this.allStringTasks = Files.readAllLines(path);
+        } catch (IOException e) {
+            System.err.println("Error in path set!");
+        }
+    }
+
+    /**
+     * returns all the tasks stored in the local directory as an arraylist
+     * 
+     * @return Arraylist of Task object 
+     * @throws FileNotFoundException if local file directory does not exist
+     * @throws DukeException if either the Todo, Event, Deadline task is not 
+     * initialized because of a wrong input given by the user
+     */
+    protected ArrayList<Task> load() throws FileNotFoundException, DukeException {
+        HashMap<String, Boolean> stringTasks = new HashMap<>();
+        for (String tasks : allStringTasks) {
+            this.storeDiskStorageInputs(tasks, stringTasks);
+        }
+        return this.changeStorageToTasks(stringTasks);
+    }
+
+    private ArrayList<Task> changeStorageToTasks(
+            HashMap<String, Boolean> stringTasks) throws DukeException {
+        ArrayList<Task> finalOutputTasks = new ArrayList<>();
+        for (Map.Entry<String, Boolean> entry : stringTasks.entrySet()) {
+            String fullInstructions = entry.getKey();
+            String[] instructionsArray = fullInstructions.split(" ");
+            String crucialInstructions = String.join(" ",
+                    Arrays.copyOfRange(instructionsArray,
+                            1, instructionsArray.length));
+            String identity = instructionsArray[0];
+            if (identity.equals(TODO_FULL_IDEN)) {
+                Task ToDoTask;
+                if (entry.getValue()) {
+                    ToDoTask = new ToDo(crucialInstructions).
+                            updateStatus(Status.COMPLETED.getStatus());
+                } else {
+                    ToDoTask = new ToDo(crucialInstructions);
+                }
+                finalOutputTasks.add(ToDoTask);
+            } else if (identity.equals(EVENT_FULL_IDEN)) {
+                Task EventTask;
+                if (entry.getValue()) {
+                    EventTask = new Event(crucialInstructions).
+                            updateStatus(Status.COMPLETED.getStatus());
+                } else {
+                    EventTask = new Event(crucialInstructions);
+                }
+                finalOutputTasks.add(EventTask);
+            } else if (identity.equals(DEADLINE_FULL_IDEN)) {
+                Task DeadlineTask;
+                if (entry.getValue()) {
+                    DeadlineTask = new Deadline(crucialInstructions).
+                            updateStatus(Status.COMPLETED.getStatus());
+                } else {
+                    DeadlineTask = new Deadline(crucialInstructions);
+                }
+                finalOutputTasks.add(DeadlineTask);
+            } else {
+                String errorMsg = "No such task!";
+                throw new DukeException(errorMsg);
+            }
+        }
+        return finalOutputTasks;
+    }
```

Comment:
This is a really long method, perhaps consider abstracting some parts of this method or removing duplicated code?